src/hooks/useExtendedComponentMap.ts

```diff
@@ -0,0 +1,22 @@
+import {useMemo} from 'react';
+
+import {CustomConfig} from '@gravity-ui/page-constructor';
+
+import componentMap from '../constructor/blocksMap';
+
+export const useExtendedComponentMap = (custom: CustomConfig | undefined) =>
+    useMemo(
+        () =>
+            custom
```

Comment:
Can we rewrite conditional part as is there https://github.com/gravity-ui/page-constructor/blob/main/src/containers/PageConstructor/PageConstructor.tsx#L68

Reply:
Sorry, but I'm not sure that I can do this. The `getCustomTypes` method returns `string[]` value that is an array of keys from custom component map. While here we should merge two component maps.